driver/src/main/java/org/neo4j/driver/ConnectionPoolMetrics.java

```diff
@@ -124,15 +120,15 @@
     long totalAcquisitionTime();
```

Comment:
I wonder if we should compute this or just delegate to a monitoring tool. Something like Prometheus or Kibana should be able to compute this, based on the available counters and gauges.

Reply:
I am not sure how to calculate this acquisition time with other existing status.
This method basically give the acquisition time. Exposing it in a totally number is just how we expose it. A user would typically be interested in average acquisition time, which I give out the code to compute in javadoc

Could you give me some an example how we could we deduce this acquisition time with other data? Monitoring tool is smart but it has to use some math to compute this value, right? How the math would looks like???

Reply:
I guess in Prometheus, this query method would do the trick:

https://prometheus.io/docs/prometheus/latest/querying/functions/#aggregation_over_time

The idea is that those tools just collect all the counters and gauges and do the aggregration etc. on that abstract level.

Reply:
No, this is the connection acquisition time. For example, during 1s, we have 4 connection acquire operation and their acquisition time are `1ms, 2ms, 1ms, 3ms`. Then we can use that tool to calculate the total acquisition time, average acquisition time etc. However there is no good way to give a user these acquisition time `1ms, 2ms, 1ms, 3ms`. 

Imagine, in order to give a user these values, then we need to give them a list. But if we give them a list then there are too many questions one can ask: How long should we keep an acquisition time in the list? How to record the time when a new number is captured? 

Thus we give them a sum of `1ms, 2ms, 1ms, 3ms`. If the user query every 1s, then the user would know that the sum has increased 7ms since last time he queried, and the total acquisition count has increased 4, so he can compute that the average of the acquisition time over 1s is around 2ms. 

I do not think a user can deduce this average acquisition time from other counters and gauges. We have gauge `acquiring` and counter `acquired`.  It is totally possible, when a user query every 1s, the `acquiring` is always 0. However the `acquired` increased by 4. I do not see a way to compute the average acquisition time.

Reply:
I see. Thanks for the explanation, that makes sense. I think that I thought there was another way to get the individual acquisition time.